cloud/spawn.go

```diff
@@ -172,6 +178,35 @@ func CreateSpawnHost(so SpawnOptions) (*host.Host, error) {
 	return intentHost, nil
 }
 
+func CheckInstanceTypeValid(ctx context.Context, d distro.Distro, requestedType string, allowedTypes []string) error {
+	if !IsEc2Provider(d.Provider) {
+		return errors.Errorf("cannot specify instance type for provider '%s'", d.Provider)
+	}
+	foundType := false
+	for _, allowedType := range allowedTypes {
```

Comment:
an alternative could be to use `util.StringSliceContains`

Reply:
Good call!